examples/core/src/main/scala/com/twitter/concurrent/AsyncSemaphore.scala

```diff
@@ -59,8 +59,8 @@ class AsyncSemaphore protected (initialPermits: Int, maxWaiters: Option[Int]) {
   // Serves as our intrinsic lock.
   private[this] final def lock: Object = waitq
 
-  private[this] val semaphorePermit: _root_.com.twitter.concurrent.Permit = new Permit {
-    private[this] val ReturnThis: _root_.com.twitter.util.Return[_root_.com.twitter.concurrent.Permit] = Return(this)
+  private[this] val semaphorePermit: _root_.scala.AnyRef with _root_.com.twitter.concurrent.Permit = new Permit {
```

Comment:
Return types produced by RscCompat directly mirror the return types inferred by Scalac. 

In this case, Scalac infers the return type as `AnyRef with Promise`, because that's the normalized representation for the type of the anonymous class on the right-hand side. 

(Due to some reason, Scalac insists that compound types start with a base _class_, so just `Promise` is not good enough because `Promise` is a trait. I had to spend a non-trivial amount of type reproducing this behavior in Rsc - https://github.com/twitter/rsc/blob/a2d3a810d31d23191730fe7be87d923000caf506/rsc/src/main/scala/rsc/semanticdb/Semanticdb.scala#L264-L327).